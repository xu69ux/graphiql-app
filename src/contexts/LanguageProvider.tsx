import { useState } from 'react';
import { LanguageContext } from './LanguageContext';

export const LanguageProvider = ({ children }) => {
  const [language, setLanguage] = useState('eng');

  return (
    <LanguageContext.Provider value={{ language, setLanguage }}>
      {children}
    </LanguageContext.Provider>
  );
};
