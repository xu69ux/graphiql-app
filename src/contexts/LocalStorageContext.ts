import React from 'react';

interface LocalStorageContextProps {
  getItem: (key: string) => string | null;
  setItem: (key: string, value: string) => void;
  removeItem: (key: string) => void;
  clear: () => void;
  isLocalCleared: boolean;
}

export const LocalStorageContext = React.createContext<
  LocalStorageContextProps | undefined
>(undefined);
