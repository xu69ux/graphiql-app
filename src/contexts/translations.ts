export const translations = {
  eng: {
    greeting: 'hello',
    login: 'log in',
    loginTitle: 'Log in',
    logout: 'log out',
    signup: 'sign up',
    signupTitle: 'Sign up',
    welcome: 'to use this awesome application, log in with a fake email',
    welcomeDescription1:
      "JS do IT is excited to present our second collaborative project — GraphiQL Playground, a powerful tool for working with GraphQL queries. GraphiQL is not just a sandbox; it's a full-fledged IDE with extended functionality, including authorization, authentication capabilities, and support for arbitrary open GraphQL endpoints.",
    welcomeDescription2:
      'JS do IT is our creative team gathered in the name of crafting inspiring projects. This marks our second step in collaborative development, and we aspire to make GraphiQL Playground not just a tool but a true piece of art in the world of GraphQL.',
    welcomeDescription3:
      'We believe that GraphiQL Playground will not only streamline your work with GraphQL queries but also serve as a splendid example of what can be achieved by combining creative potential and technical expertise.',
    welcomeDescription4:
      "Wishing you vibrant ideas, effective teamwork, and the joy of creating something extraordinary! Don't forget about inspiration and creativity in every line of code. Best of luck, JS do IT!",
    or: 'or',
    madeWith: 'made with',
    by: 'by',
    email: 'email',
    password: 'password',
    username: 'username',
    noAccount: 'No account? Then, ',
    yesAccount: 'Have an account? Then, ',
    tooltipUsername: 'please use any username',
    tooltipEmail: 'please use any email',
    tooltipEmailLogin: 'enter the email provided during registration',
    tooltipPassword: 'must be at least 8 characters',
    tooltipPasswordLogin: 'enter the password provided during registration',
    emailRequired: 'Email is required',
    emailInvalid: 'Invalid email',
    passwordRequired: 'Password is required',
    passwordInvalid: 'Password must meet the strength criteria',
    passwordInvalidCriteria: 'Password must have: ',
    passwordInvalidCriteriaLowercase: 'at least one lowercase letter',
    passwordInvalidCriteriaUppercase: 'at least one uppercase letter',
    passwordInvalidCriteriaNumber: 'at least one number',
    passwordInvalidCriteriaSpecial: 'at least one special character',
    passwordInvalidCriteriaLength: 'at least 8 characters long',
    commonErrorBody: 'Something went wrong. Please try again later.',
    regAlreadyExistBody: 'An account with this email already exists.',
    loginUserNotFoundBody:
      'The user does not exist or the email/password is incorrect.',
    logOutSuccesBody: 'Log out successful!',
    notFoundTitle1: "oops, you've ended up somewhere you didn't mean to!",
    notFoundTitle2: 'go back to the ',
    notFoundTitle3: 'or check out a random meme:',
    mainPage: 'main page ',
  },
  rus: {
    greeting: 'привет',
    login: 'войти',
    loginTitle: 'Войти',
    logout: 'выйти',
    signup: 'зарегистрироваться',
    signupTitle: 'Зарегистрироваться',
    welcome:
      'чтобы использовать это офигенное приложение, войдите с помощью фейковой почты',
    welcomeDescription1:
      'JS do IT рад представить вам свой второй совместный проект — GraphiQL Playground, мощный инструмент для работы с GraphQL-запросами. GraphiQL — это не просто песочница, а целый IDE, расширенный функционал которого включает в себя возможности авторизации, аутентификации и поддержку произвольных открытых GraphQL-эндпоинтов.',
    welcomeDescription2:
      'JS do IT - это наша творческая команда, собравшаяся во имя создания вдохновляющих проектов. Это уже наш второй шаг в совместной разработке, и мы стремимся сделать GraphiQL Playground не просто инструментом, а настоящим искусством в мире GraphQL.',
    welcomeDescription3:
      'Мы уверены, что GraphiQL Playground не только облегчит вам работу с GraphQL-запросами, но и станет прекрасным примером того, что можно достичь, объединив творческий потенциал и техническое мастерство.',
    welcomeDescription4:
      'Желаем вам ярких идей, эффективной работы в команде и удовольствия от создания чего-то удивительного! Не забывайте о вдохновении и креативности в каждой строчке кода. Удачи, JS do IT!',
    or: 'или',
    madeWith: 'сделано с',
    by: '',
    email: 'почта',
    password: 'пароль',
    username: 'имя пользователя',
    noAccount: 'Нет аккаунта? Тогда, ',
    yesAccount: 'Есть аккаунт? Тогда, ',
    tooltipUsername: 'пожалуйста, используйте любое имя пользователя',
    tooltipEmail: 'пожалуйста, используйте любую почту',
    tooltipEmailLogin: 'введите почту, указанную при регистрации',
    tooltipPassword: 'должен быть не менее 8 символов',
    tooltipPasswordLogin: 'введите пароль, указанный при регистрации',
    emailRequired: 'Почта обязательна',
    emailInvalid: 'Неверная почта',
    passwordRequired: 'Пароль обязателен',
    passwordInvalid: 'Пароль должен соответствовать критериям надежности',
    passwordInvalidCriteria: 'Пароль должен иметь: ',
    passwordInvalidCriteriaLowercase: 'хотя бы одну строчную букву',
    passwordInvalidCriteriaUppercase: 'хотя бы одну заглавную букву',
    passwordInvalidCriteriaNumber: 'хотя бы одну цифру',
    passwordInvalidCriteriaSpecial: 'хотя бы один спецсимвол',
    passwordInvalidCriteriaLength: 'хотя бы 8 символов',
    commonErrorBody:
      'Что-то пошло не так. Пожалуйста, повторите попытку позже.',
    regAlreadyExistBody:
      'Учетная запись с такой электронной почтой уже создана.',
    loginUserNotFoundBody:
      'Такого пользователя не существует или неверно указан адрес электронной почты/пароль.',
    logOutSuccesBody: 'Выход из учетной записи успешный!',
    notFoundTitle1: 'упс, похоже, вы попали не туда!',
    notFoundTitle2: 'вернитесь на ',
    notFoundTitle3: 'или посмотрите случайный мем:',
    mainPage: 'главную страницу ',
  },
  ukr: {
    greeting: 'привіт',
    login: 'увійти',
    loginTitle: 'Увійти',
    logout: 'вийти',
    signup: 'зареєструватися',
    signupTitle: 'Зареєструватися',
    welcome:
      'щоб користуватися цим офігенним додатком, увійдіть за допомогою фейкової пошти',
    welcomeDescription1:
      'JS do IT має велику радість представити вам наш другий спільний проект — GraphiQL Playground, потужний інструмент для роботи з GraphQL-запитами. GraphiQL — це не просто пісочниця; це повноцінне Інтегроване середовище розробки (IDE) з розширеним функціоналом, що включає в себе можливості авторизації, аутентифікації та підтримку довільних відкритих GraphQL-кінцевих точок.',
    welcomeDescription2:
      "JS do IT - це наша творча команда, яка об'єдналася в ім'я створення надихаючих проектів. Це вже наш другий крок у спільній розробці, і ми прагнемо зробити GraphiQL Playground не лише інструментом, але справжнім мистецтвом у світі GraphQL.",
    welcomeDescription3:
      'Ми впевнені, що GraphiQL Playground не лише полегшить вашу роботу з GraphQL-запитами, але й стане чудовим прикладом того, що можна досягти, поєднуючи творчий потенціал і технічну майстерність.',
    welcomeDescription4:
      'Бажаємо вам яскравих ідей, ефективної роботи в команді та задоволення від створення чогось неймовірного! Не забувайте про натхнення та креативність в кожному рядку коду. Удачі, JS do IT!',
    or: 'або',
    madeWith: 'зроблено з',
    by: '',
    email: 'пошта',
    password: 'пароль',
    username: "ім'я користувача",
    noAccount: 'Відсутній аккаунт? Тоді, ',
    yesAccount: 'Є аккаунт? Тоді, ',
    tooltipUsername: 'будь ласка, вкажіть будь-яке ім’я користувача',
    tooltipEmail: 'будь ласка, вкажіть будь-яку пошту',
    tooltipEmailLogin: 'введіть пошту, вказану при реєстрації',
    tooltipPassword: 'повинен бути не менше 8 символів',
    tooltipPasswordLogin: 'введіть пароль, вказаний при реєстрації',
    emailRequired: 'Пошта обов’язкова',
    emailInvalid: 'Невірна пошта',
    passwordRequired: 'Пароль обов’язковий',
    passwordInvalid: 'Пароль повинен відповідати критеріям надійності',
    passwordInvalidCriteria: 'Пароль повинен мати: ',
    passwordInvalidCriteriaLowercase: 'хоча б одну малу літеру',
    passwordInvalidCriteriaUppercase: 'хоча б одну велику літеру',
    passwordInvalidCriteriaNumber: 'хоча б одну цифру',
    passwordInvalidCriteriaSpecial: 'хоча б один спецсимвол',
    passwordInvalidCriteriaLength: 'хоча б 8 символів',
    commonErrorBody: 'Щось пішло не так. Будь ласка, спробуйте пізніше.',
    regAlreadyExistBody: 'Обліковий запис з цим email вже існує.',
    loginUserNotFoundBody:
      'Такого користувача не існує або email/пароль невірні.',
    logOutSuccesBody: 'Успішний вихід з облікового запису!',
    notFoundTitle1: 'ой, здається, ви потрапили кудись не туди!',
    notFoundTitle2: 'поверніться на ',
    notFoundTitle3: 'або перегляньте випадковий мем:',
    mainPage: 'головну сторінку ',
  },
};
