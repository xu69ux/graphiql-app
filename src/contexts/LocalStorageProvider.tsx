import { useState } from 'react';
import { LocalStorageContext } from '@contexts/LocalStorageContext';

export const LocalStorageProvider = ({ children }) => {
  const [isLocalCleared, setLocalCleared] = useState(true);

  const getItem = (key: string) => localStorage.getItem(key);
  const setItem = (key: string, value: string) => {
    localStorage.setItem(key, value);
    setLocalCleared(false);
  };
  const removeItem = (key: string) => localStorage.removeItem(key);
  const clear = () => {
    localStorage.clear();
    setLocalCleared(true);
  };

  return (
    <LocalStorageContext.Provider
      value={{ getItem, setItem, removeItem, clear, isLocalCleared }}
    >
      {children}
    </LocalStorageContext.Provider>
  );
};
