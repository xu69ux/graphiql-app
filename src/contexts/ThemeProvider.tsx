import { useState, useEffect } from 'react';
import { ThemeContext } from './ThemeContext';
import { translations } from './translations';
import useLanguage from '../hooks/useLanguage';

export const ThemeProvider = ({ children }) => {
  const { language } = useLanguage();
  const darkTheme = translations?.[language].dark;
  const crazyTheme = translations?.[language].crazy;
  const [theme, setTheme] = useState(darkTheme);

  useEffect(() => {
    setTheme(darkTheme ? darkTheme : crazyTheme);
  }, [language, darkTheme, crazyTheme]);

  const toggleTheme = () => {
    setTheme((prevTheme) =>
      prevTheme === crazyTheme ? darkTheme : crazyTheme,
    );
  };

  useEffect(() => {
    document.body.dataset.theme = theme;
  }, [theme]);

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};
