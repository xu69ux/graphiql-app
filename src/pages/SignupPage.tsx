import { useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { FormSignUp } from '../components';
import { translations } from '../contexts/translations';
import useLanguage from '../hooks/useLanguage';

import '@styles/Auth.css';

const SignupPage = () => {
  const navigate = useNavigate();
  const userIs = sessionStorage.getItem('authInfo');
  const { language } = useLanguage();

  useEffect(() => {
    if (userIs) navigate('/graphiql');
  }, [userIs, navigate]);

  return (
    !userIs && (
      <div className='auth-container'>
        <h1 className='auth-title' data-testid='signup-title'>
          {translations?.[language]?.signupTitle}
        </h1>
        <FormSignUp />
        <p className='no-account'>
          {translations?.[language]?.yesAccount}
          <Link to='/login' className='login-link' data-testid='login-link'>
            {translations?.[language]?.login}
          </Link>
          !
        </p>
      </div>
    )
  );
};

export default SignupPage;
