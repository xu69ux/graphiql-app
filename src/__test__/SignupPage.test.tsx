import { act, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { LanguageContext } from '../contexts/LanguageContext';
import SignupPage from '../pages/SignupPage';

describe('SignUpPage component', () => {
  let container: HTMLElement;
  test('renders correctly', async () => {
    await act(
      async () =>
        ({ container } = render(
          <Router>
            <LanguageContext.Provider
              value={{ language: 'eng', setLanguage: () => {} }}
            >
              <SignupPage />
            </LanguageContext.Provider>
          </Router>,
        )),
    );

    expect(container).toMatchSnapshot();
  });

  it('renders sign up title', async () => {
    await act(async () => {
      render(
        <LanguageContext.Provider
          value={{ language: 'eng', setLanguage: () => {} }}
        >
          <Router>
            <SignupPage />
          </Router>
        </LanguageContext.Provider>,
      );
    });

    expect(screen.getByTestId('signup-title')).toBeInTheDocument();
  });

  it('renders sign up form', async () => {
    await act(async () => {
      render(
        <LanguageContext.Provider
          value={{ language: 'eng', setLanguage: () => {} }}
        >
          <Router>
            <SignupPage />
          </Router>
        </LanguageContext.Provider>,
      );
    });

    expect(screen.getByPlaceholderText(/Username/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Password/i)).toBeInTheDocument();
  });

  test('renders login link', async () => {
    await act(async () => {
      render(
        <Router>
          <LanguageContext.Provider
            value={{ language: 'eng', setLanguage: () => {} }}
          >
            <SignupPage />
          </LanguageContext.Provider>
        </Router>,
      );
    });
    const linkElement = screen.getByTestId('login-link');
    expect(linkElement).toBeInTheDocument();
  });
});
