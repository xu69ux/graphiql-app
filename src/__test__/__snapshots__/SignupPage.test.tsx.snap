// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage component renders correctly 1`] = `
<div>
  <div
    class="auth-container"
  >
    <h1
      class="auth-title"
      data-testid="signup-title"
    >
      Signup
    </h1>
    <form>
      <div
        class="signup"
      >
        <div
          class="input-wrapper"
        >
          <input
            name="username"
            placeholder="username"
            type="text"
          />
        </div>
        <div
          class="error"
        />
        <div
          class="input-wrapper"
        >
          <input
            name="email"
            placeholder="email"
            type="text"
          />
        </div>
        <div
          class="error"
        />
        <div
          class="input-wrapper"
        >
          <input
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="password-indicator"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0;"
          >
            <div
              style="width: 0%; height: 100%; opacity: 0;"
            />
          </div>
        </div>
        <div
          class="error"
        />
        <button
          class="btn btn-sign"
          type="submit"
        >
          Signup
        </button>
      </div>
    </form>
    <p
      class="no-account"
    >
      Have an account? Then, 
      <a
        class="login-link"
        data-testid="login-link"
        href="/login"
      >
        login
      </a>
      !
    </p>
  </div>
</div>
`;
